lib/Conversion/ATenToTCF/CoreOpConversionPatterns.cpp

```diff
@@ -68,10 +68,49 @@ class ConvertBinaryElementwise : public OpRewritePattern<SourceOp> {
 
 /// The ATen Conv2dOp has seven arguments:
 ///   input, weight, bias, stride, padding, dilation, groups
+
 class ConvertATenConv2d : public OpRewritePattern<aten::Conv2dOp> {
   using OpRewritePattern::OpRewritePattern;
   LogicalResult matchAndRewrite(aten::Conv2dOp srcConv2dOp,
                                 PatternRewriter &rewriter) const override {
+    auto results = srcConv2dOp.getOperation()->getResults();
+    assert(srcConv2dOp.getNumOperands() == 7 && "expected seven (7) operands");
+    assert(results.size() == 1 && "expected single result op");
+    auto *strideOp = srcConv2dOp.stride().getDefiningOp();
```

Comment:
can you use `getDefiningOp<TheOp>()` and bail out if it is not the type you expect (basicpy.build_list, right?)

Reply:
Matching the `basicpy.build_list` op is too specific where we're expecting an int-list, which are constant values for the time being. Although, my code is also too specific in that it matches a source-op whose inputs are constant-int operands. Is there a way to just say match an int-list operand, or does an int-list always imply a `basicpy.build_list` source-op?

Reply:
Specificity is what we want here. `basicpy.build_list` is the only op we can currently safely match here to discover if the list is the constant we want. We cannot assume that every op with two constant operands that produces a list, must produce a list of two elements consisting of the two operands.

Reply:
Specificity is what we want here. `basicpy.build_list` is the only op we can safely handle here. We cannot assume that every op that produces a list and has two constant operands necessarily produces a list consisting of those two operands.